soda_machine.rb

```diff
@@ -6,13 +6,33 @@ def initialize(args = {})
     @cash = args[:cash]
   end
 
+  #the sodas are initialized as an array, so I have to simply find the
+  #length of the array containing the soda objects
   def current_inventory_count
+    @sodas.length
   end
 
+  #returns the soda object that the user request
+  #which are stored in the array of sodas, so I have to search the array for
+  #the soda object
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        return soda
+      end
+    end
+    return nil
   end
 
   def sell(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
```

Comment:
This code looks awfully similar to the code in `find_soda` -- how could you DRY this up?